test/extensions/filters/common/expr/expr_context_speed_test.cc

```diff
@@ -0,0 +1,171 @@
+#include "source/common/http/header_map_impl.h"
+#include "source/common/network/address_impl.h"
+#include "source/common/protobuf/protobuf.h"
+#include "source/common/router/string_accessor_impl.h"
+#include "source/extensions/filters/common/expr/context.h"
+
+#include "test/mocks/local_info/mocks.h"
+#include "test/mocks/ssl/mocks.h"
+#include "test/mocks/stream_info/mocks.h"
+#include "test/mocks/upstream/host.h"
+#include "test/test_common/utility.h"
+
+#include "benchmark/benchmark.h"
+#include "gmock/gmock.h"
+
+namespace Envoy {
+
+using testing::NiceMock;
+using testing::Return;
+
+class ExpressionContextSpeedTest {
+public:
+  ExpressionContextSpeedTest(size_t num_attributes) {
+    ssl_info_ = std::make_shared<NiceMock<Ssl::MockConnectionInfo>>();
```

Comment:
nit: mocks are generally somewhat slow, so it's better to use real data structures for performance measurements.

Reply:
@kyessenov I tried doing it but couldn't get it to work. The benchmark test shouldn't be blocking so if it's okay, I can try to take a look at it separately.

Reply:
Yes, it's fine since a map lookup is probably roughly mock virtual function table lookup, so results are not dominated by mocks overhead.